client/flags/flags.go

```diff
@@ -111,6 +114,20 @@ func AddTxFlagsToCmd(cmd *cobra.Command) {
 	cmd.SetOut(cmd.OutOrStdout())
 }
 
+// GetKeyringDir finds the directory for the keyring storage
+func GetKeyringDir(pflags *pflag.FlagSet, homeDir string) (string, error) {
+	keyringDir, err := pflags.GetString(FlagKeyringDir)
+	if keyringDir == "" {
+		// Use the specified home directory.
+		keyringDir = homeDir
+	}
+	if keyringDir == "" {
+		// Default to the node's home directory.
+		keyringDir, err = pflags.GetString(FlagHome)
+	}
+	return keyringDir, err
+}
+
```

Comment:
This is a bit confusing to grok IMHO. We know that when this is called, `clientCtx.HomeDir` is populated. So really, this method should be a simple if/else.

```go
if keyringDir != "" {
  return keyringDir
}

return homeDir
```

Reply:
Now that I think of it, I'm not even sure we need a function for this -- it's so concise.

Reply:
> This is a bit confusing to grok IMHO. We know that when this is called, `clientCtx.HomeDir` is populated. So really, this method should be a simple if/else.

All the `gaiad keys ...` commands don't have a `clientCtx.HomeDir` and so need to handle the `FlagHome` explicitly.  I will inline the details in all the appropriate places, though, in the interest of understandability.